src/v2.rs

```diff
@@ -340,22 +336,8 @@ impl EszipV2 {
       source_map: EszipV2SourceSlot::Ready(Arc::new(vec![])),
     };
     let mut modules = self.modules.lock().unwrap();
-    if modules.get(&specifier).is_none() {
-      modules.insert(specifier.clone(), module);
-    }
-    let mut module_ordering =
-      Vec::with_capacity(self.ordered_modules.len() + 1);
-    module_ordering.push(specifier.clone());
-    let old_module_ordering =
-      std::mem::replace(&mut self.ordered_modules, module_ordering);
-    // `old_module_ordering` might contain the same module as the import map
-    // that the given `specifier` specifies. To avoid having duplicate module,
-    // we check and exclude that from `old_module_ordering`.
-    let old_module_ordering_without_import_map =
-      old_module_ordering.into_iter().filter(|x| x != &specifier);
-    self
-      .ordered_modules
-      .extend(old_module_ordering_without_import_map);
+    modules.insert(specifier.clone(), module);
+    modules.to_front(&specifier);
```

Comment:
For reviewers: `to_front` is important in order to place the import map at the top of modules. I use [hashlink](https://docs.rs/hashlink/0.8.1/hashlink/index.html) crate instead of [linked_hash_map](https://docs.rs/linked-hash-map/latest/linked_hash_map/), because linked_hash_map doesn't provide a method similar to `to_front`.